### ISSUE TITLE
Allow `query: from(... )

### ISSUE THREAD

**Issue by lloydtabb** [looker-open-source/malloy#149]
A pretty good use case for this is joining the results of one query with another without having to create an explore.

```
query: from(aircraft_models->{
  group_by: m is manufacturer
  aggregate: num_models is count(*)
  }){
  join: seats is from(
    aircraft_models->{
      group_by: m is manufacturer
      aggregate: total_seats is seats.sum()
    }
  ) on m
}
-> {
  project: [
    m
    num_models
    seats.total_seats
  ]
  order_by: 2 desc
  limit: 1
}
```

**Comment by mtoy-googly-moogly** [comment]
That think you wrote is supposed to work. If it doesn't let me know.

**Comment by mtoy-googly-moogly** [comment]
ok, it doesn't work and this just goes on the someday pile, it was supposed to, but it fell through the cracks.

**Comment by lloydtabb** [comment]
This is a pretty useful construct for building funnels.  Here is a super simple case (and there is a test in 'nomodel.spec.ts')

```
        query: from(
          table('malloytest.state_facts)->{group_by:state}
          ) {
            measure: state_count is count(distinct state)
          }
        -> {
          aggregate: state_count
        }
```

**Comment by mtoy-googly-moogly** [comment]
Fixed 485eff30e4e40798cbe31ef11a5fbd130aeefc9e
